spring-boot-starter/starter-security/src/test/java/org/camunda/bpm/spring/boot/starter/security/oauth2/CamundaBpmSecurityAutoConfigOauth2ApplicationIT.java

```diff
@@ -0,0 +1,173 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.authentication;
+
+@AutoConfigureMockMvc
+@TestPropertySource("/oauth2-mock.properties")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class CamundaBpmSecurityAutoConfigOauth2ApplicationIT extends AbstractSpringSecurityIT {
+
+  protected static final String UNAUTHORIZED_USER = "mary";
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Autowired
+  private FilterRegistrationBean<Filter> filterRegistrationBean;
+
+  @Autowired
+  private ClientRegistrationRepository registrations;
+
+  @MockBean
+  private OAuth2AuthorizedClientService authorizedClientService;
+
+  @Rule
+  public ProcessEngineLoggingRule loggingRule = new ProcessEngineLoggingRule().watch(AuthorizeTokenFilter.class.getCanonicalName());
+
+  private OAuth2AuthenticationProvider spiedAuthenticationProvider;
+
+  @Before
+  public void setup() throws Exception {
+    super.setup();
+    spyAuthenticationProvider();
+  }
+
+  @Test
+  public void testSpringSecurityAutoConfigurationCorrectlySet() {
+    assertThat(getBeanForClass(CamundaSpringSecurityOAuth2AutoConfiguration.class, mockMvc.getDispatcherServlet().getWebApplicationContext())).isNotNull();
+    assertThat(getBeanForClass(CamundaBpmSpringSecurityDisableAutoConfiguration.class, mockMvc.getDispatcherServlet().getWebApplicationContext())).isNull();
+  }
+
+  @Test
+  public void testWebappWithoutAuthentication() throws Exception {
+    // given no authentication
+
+    // when
+    mockMvc.perform(MockMvcRequestBuilders.get(baseUrl + "/camunda/api/engine/engine/default/user")
+            .accept(MediaType.APPLICATION_JSON))
+        .andDo(MockMvcResultHandlers.print())
+        // then
+        .andExpect(MockMvcResultMatchers.status().isFound())
```

Comment:
Yes indeed, it might not be obvious why the server returns 302, but it basically depends on our configuration of oauth2 for the camunda webapps. I agree that I could improve the `// then` comment to make it more explicit, although I'd rather keep the test name as is to avoid it being to verbose.